src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java

```diff
@@ -196,6 +196,56 @@ public static GATKRead createReadAlignedToRef(final GATKRead originalRead,
         return Arrays.copyOfRange(bases, basesStart, basesStop + 1);
     }
 
+    public static byte[] getBasesAlignedOneToOne(final GATKRead read) {
+        return getSequenceAlignedOneToOne(read, true, (byte)'N');
+    }
+
+    public static byte[] getBaseQualsAlignedOneToOne(final GATKRead read) {
+        return getSequenceAlignedOneToOne(read, false, (byte)0);
+    }
+
+    public static byte[] getSequenceAlignedOneToOne(final GATKRead read, final boolean forBases, final byte padWith) {
+        final Cigar cigar = read.getCigar();
+        if (!cigar.containsOperator(CigarOperator.DELETION) && !cigar.containsOperator(CigarOperator.INSERTION)) {
+            if (forBases) {
+                return read.getBasesNoCopy();
+            }
+            else {
+                return read.getBaseQualitiesNoCopy();
+            }
+        }
+        else {
+            final byte[] literalBases = forBases? read.getBasesNoCopy() : read.getBaseQualitiesNoCopy();
+            final byte[] paddedBases = new byte[CigarUtils.countRefBasesBasedOnCigarIgnoringHardClips(read, 0, cigar.numCigarElements())];
+            int literalPos = 0;
+            int paddedPos = 0;
+            for ( int i = 0; i < cigar.numCigarElements(); i++ ) {
+                final CigarElement ce = cigar.getCigarElement(i);
+                switch ( ce.getOperator() ) {
```

Comment:
This switch can be rewriten with a couple of ifs and using the ```consumesReadBases``` and ```consumesReferenceBases``` without the need for list operators explicitly.